tests/data/py_38/pep_572_remove_parens.py

```diff
@@ -103,3 +123,23 @@ def a():
 def this_is_so_dumb() -> (please := no):
     pass
 
+
+async def await_the_walrus():
+    with (x := y):
+        pass
+
+    with (x := y) as z, (a := b) as c:
+        pass
+
+    with (x := await y):
+        pass
+
+    with (x := await a, y := await b):
+        pass
+
+    # Ideally we should remove one set of parentheses
+    with ((x := await a, y := await b)):
+        pass
```

Comment:
That seems fine, since the main purpose of this test is to make sure we don't crash on this case. We can always change the exact test output when that part of the preview style becomes stable.